x-pack/plugins/security/server/saved_objects/audit_events.ts

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { AuditEventDecorator } from '../../../../../src/core/server';
+
+export interface SavedObjectEventArgs {
+  action: string;
```

Comment:
The one thing I'm not in love with here is the need to pass in the `action` to the already named audit events. I understand why you structured it this way though, since `savedObjectCreateEvent ` (for example) is used for both the regular "create" method and the "bulk_create" method. It seems silly to duplicate the entire event here just to avoid having to pass the `action` through.

I don't have any other suggestions to offer though, so feel free to ignore this. My only alternative was to narrow the type of `action` to just the expected action strings that we plan on passing through, but that likely doesn't buy us much.

Reply:
I'm not in love with the approach either. There's a bit of duplication still that can be removed and like you say the way events are broken down into individual functions by CRUD operation are slightly arbitrary.

I think this works as a starting point but is definitely worth revisiting once we implement a few other events (e.g. http request or user / role management) to see what pattern works best.